modules/users/tests/server/user.server.routes.tests.js

```diff
@@ -1086,6 +1086,118 @@ describe('User CRUD tests', function () {
     });
   });
 
+  describe('Username Update', function () {
+    it('should update it', function (done) {
+      var user2 = _user;
+      user2.username = _user.username + '01';
+      delete user2.email;
+      agent.post('/api/auth/signin')
+        .send(credentials)
+        .expect(200)
+        .end(function (err) {
+          if (err) {
+            return done(err);
+          }
+          agent.put('/api/users')
+            .send(user2)
+            .expect(200)
+            .end(function (err, res) {
+              if (err) {
+                return done(err);
+              }
+              res.body.username.should.equal(user2.username);
```

Comment:
We could also test whether the `displayUsername` was updated.

Additionally we could test that the username is changed in the database, too.